docs/plugins/reindex.asciidoc

```diff
@@ -0,0 +1,608 @@
+
+These APIs are brothers so they live in the same plugin. Both use
+{ref}/search-request-scroll.html[Scroll] and {ref}/docs-bulk.html[Bulk] APIs
+to send an index request per document though they may not work that way in the
+future. Regardless of how they work internally they will always work **as**
+**though** they send an index request per document that matches a scroll.
+
+[float]
+==== Installation
+
+This plugin can be installed using the plugin manager:
+
+[source,sh]
+----------------------------------------------------------------
+sudo bin/plugin install reindex
+----------------------------------------------------------------
+
+The plugin must be installed on every node in the cluster, and each node must
+be restarted after installation.
+
+[float]
+==== Removal
+
+The plugin can be removed with the following command:
+
+[source,sh]
+----------------------------------------------------------------
+sudo bin/plugin remove reindex
+----------------------------------------------------------------
+
+The node must be stopped before removing the plugin.
+
+[[update-by-query-usage]]
+==== Using `_update_by_query`
+
+The simplest usage of `_update_by_query` just `touch`es every document in an
+index:
+
+[source,js]
+--------------------------------------------------
+POST /twitter/_update_by_query
+--------------------------------------------------
+// AUTOSENSE
+
+That will return something like this:
+
+[source,js]
+--------------------------------------------------
+{
+  "took" : 639,
+  "updated": 1235,
+  "batches": 13,
+  "version_conflicts": 2,
+  "failures" : [ ]
+}
+--------------------------------------------------
+
+`_update_by_query` gets a snapshot of the index when it starts and indexes what
+it finds using `internal` versioning. That means that if the document changes
+between the time when the snapshot was taken and when the index request is
+processed then you'll get a version conflict. When the versions match the
+document is changed and the version number is incremented.
+
+Version conflicts are just counted and not returned by default. To count them
+you must set `conflicts=abort` on the url or `"conflicts": "abort"` in the
+request body. If you do then the `failures` element will contain version
+conflict information. There is no `conflicts=record` because we number of
+conflicts could grow without bounds so its either `abort` or nothing.
+
+Speaking of the `failures` element, it will contain a list of failures if
+there are any. The only "optional" sort of failures are version conflicts. All
+other failures always cause the operation to abort. While the first failure
+causes the abort, it's possible to encounter lots of subsequent failures due to
+batching so all failures are reported. Just in case you want them.
+
+It's important to remember that "abort" here doesn't mean "rollback". This
+operation isn't atomic. If it fails part way through then its changes have been
+made.
```

Comment:
I'm reworking these last few paragraphs as you suggest and will push in a few minutes.